app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java

```diff
@@ -458,7 +467,17 @@ public void onFabClick(View view, boolean needsSearch) {
             for (String id : optionSetIds) {
                 queryData.remove(id);
             }
-            queryProcessor.onNext(queryData);
+            if (selectedProgram != null && selectedProgram.displayFrontPageList()) {
+                if (queryData.size() < selectedProgram.minAttributesRequiredToSearch()) {
+                    showList = false;
+                    this.view.setFabIcon(true);
+                    queryProcessor.onNext(new HashMap<>());
+                } else {
+                    showList = true;
+                    this.view.setFabIcon(false);
+                    queryProcessor.onNext(queryData);
+                }
+            }
         }
```

Comment:
It is missing when the first condition is not met. Will fix it

Reply:
possible to add unit tests :D ?

Reply:
done!